sync_diff_inspector/source/tidb.go

```diff
@@ -126,7 +127,32 @@ func (s *TiDBSource) GetCountAndMd5(ctx context.Context, tableRange *splitter.Ra
 	chunk := tableRange.GetChunk()
 
 	matchSource := getMatchSource(s.sourceTableMap, table)
-	count, checksum, err := utils.GetCountAndMd5Checksum(ctx, s.dbConn, matchSource.OriginSchema, matchSource.OriginTable, table.Info, chunk.Where, chunk.Args)
+	indexHint := ""
+	if s.sqlHint == "auto" && len(chunk.IndexColumnNames) > 0 {
+		// If sqlHint is set to "auto" and there are index column names in the chunk,
+		// we attempt to find a matching index in the source table to use as an index hint.
+		// This is necessary because the index name might differ between the source and target tables,
+		// even if the index columns are the same. For example:
+		// 		Source: idx1(c1, c2)
+		// 		Target: idx2(c1, c2)
+		// In this case, we use the column names [c1, c2] from idx1 to find the corresponding index name (idx2) in the source table.
+		if tableInfos, err := s.GetSourceStructInfo(ctx, tableRange.GetTableIndex()); err == nil {
+			for _, index := range dbutil.FindAllIndex(tableInfos[0]) {
+				if utils.IsIndexMatchingColumns(index, chunk.IndexColumnNames) {
+					indexHint = fmt.Sprintf("/*+ USE_INDEX(`%s`.`%s`, `%s`) */",
+						matchSource.OriginSchema,
+						matchSource.OriginTable,
```

Comment:
still not changed to use `dbutil.TableName`. if your schema or table name contained a `` ` `` you could sql-inject it.